opentelemetry-sdk/src/logs/log_processor.rs

```diff
@@ -309,8 +312,13 @@ where
     if batch.is_empty() {
         return Ok(());
     }
+    // Convert the Vec<&LogData> to Vec<(&LogRecord, &InstrumentationLibrary)>
+    let export_batch = batch
+        .iter()
+        .map(|log_data| (&log_data.record, &log_data.instrumentation))
+        .collect();
```

Comment:
BatchProcessor is now doing an extra cloning - 1st one occurs when it receives the `&mut LogData`, so it can save it to channel. And its drained to a Vec! later, which was passed as-is to Exporter previously. Now another Vec! needs to be created to pass on to exporter.
Something we can revisit and address in the future.

Reply:
This operation is only creating references and collecting them, no cloning of data is happening.

Reply:
Isn't the collect making a new Vec! ?

Reply:
Oh yes, I was talking specifically about cloning. Need to revisit this later.